conans/client/cmd/copy.py

```diff
@@ -7,25 +7,13 @@
 from conans.util.files import rmdir
 
 
-def _prepare_sources(cache, ref, remote_manager, loader, remotes):
-    conan_file_path = cache.package_layout(ref).conanfile()
+def _prepare_sources(cache, layout, ref, remote_manager, loader, remotes):
+    conan_file_path = layout.conanfile()
```

Comment:
I've taken a look at the `complete_recipe_sources` function and it would be nice to pass the `layout` instead of the `cache+ref` 😉

Reply:
And then `_prepare_sources` won't need the `cache` \o/